### PR TITLE
executor: add ru details in slow log and INFORMATION_SCHEMA.SLOW_QUERY

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/42892 ref https://github.com/pingcap/tidb/issues/49318

Problem Summary:

### What changed and how does it work?
Add `Resource_group`, `Request_unit_read`, `Request_unit_write` and `Time_queued_by_rc` in slow_log and INFORMATION_SCHEMA.SLOW_QUERY.

NOTE: The three new fields are collected in client-go resourceControlInterceptor after https://github.com/tikv/client-go/pull/1070

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -271,6 +271,10 @@ github.com/ghemawat/stream v0.0.0-20171120220530-696b145b53b9/go.mod h1:106OIgoo
 github.com/ghodss/yaml v1.0.0/go.mod h1:4dBDuWmgqj2HViK6kFavaiC9ZROes6MMH2rRYeMEF04=
 github.com/gin-contrib/sse v0.0.0-20190301062529-5545eab6dad3/go.mod h1:VJ0WA2NBN22VlZ2dKZQPAPnyWw5XTlK1KymzLKsr59s=
 github.com/gin-gonic/gin v1.4.0/go.mod h1:OW2EZn3DO8Ln9oIKOvM++LBO+5UPHJJDH72/q/3rZdM=
+github.com/glorv/client-go/v2 v2.0.1-0.20231130110809-812d6876b43f h1:pMnYb5jUkLB9QqXxitBtcg/8Hne+5oYTW9P9w3UN2Jo=
```
Comment (hawkingrei) [inline go.sum:274]: Please run ```make bazel_prepare``` and upload the dependencies.

**Comment by glorv** [comment]
test failure is caused by #49113

**Comment by glorv** [comment]
@nolouch @CabinfeverB @HuSharp PTAL

File: pkg/sessionctx/variable/session.go [hunk 2]
```diff
@@ -3113,6 +3113,14 @@ const (
...
 	// SlowLogIsSyncStatsFailed is used to indicate whether any failure happen during sync stats
 	SlowLogIsSyncStatsFailed = "IsSyncStatsFailed"
+	// SlowLogResourceGroup is the resource group name that the current session bind.
+	SlowLogResourceGroup = "Resource_group"
+	// SlowLogRRU is the read request_unit(RU) cost
+	SlowLogRRU = "Request_unit_read"
+	// SlowLogWRU is the write request_unit(RU) cost
+	SlowLogWRU = "Request_unit_write"
+	// SlowLogWaitRUDuration is the total duration for kv requests to wait available request-units.
+	SlowLogWaitRUDuration = "Time_queued_by_rc"
```
Comment (nolouch) [inline pkg/sessionctx/variable/session.go:3123]: maybe we need to add tikv side wait duration in the future.

**Review by nolouch — APPROVED** [review]
LGTM

File: pkg/sessionctx/variable/session.go [hunk 3]
```diff
@@ -3365,6 +3377,20 @@ func (s *SessionVars) SlowLogFormat(logItems *SlowQueryLogItems) string {
...
+		writeSlowLogItem(&buf, SlowLogResourceGroup, logItems.ResourceGroupName)
+	}
+	if logItems.RRU > 0.0 {
+		writeSlowLogItem(&buf, SlowLogRRU, strconv.FormatFloat(logItems.RRU, 'f', -1, 64))
+	}
+	if logItems.WRU > 0.0 {
+		writeSlowLogItem(&buf, SlowLogWRU, strconv.FormatFloat(logItems.WRU, 'f', -1, 64))
+	}
+	if logItems.WaitRUDuration > time.Duration(0) {
+		writeSlowLogItem(&buf, SlowLogWaitRUDuration, strconv.FormatFloat(logItems.KVTotal.Seconds(), 'f', -1, 64))
```
Comment (crazycs520) [inline pkg/sessionctx/variable/session.go:3391]:
```suggestion
		writeSlowLogItem(&buf, SlowLogWaitRUDuration, strconv.FormatFloat(logItems.WaitRUDuration.Seconds(), 'f', -1, 64))
```

File: pkg/sessionctx/variable/session_test.go [hunk 4]
```diff
@@ -258,7 +258,11 @@ func TestSlowLogFormat(t *testing.T) {
 # Succ: true
 # IsExplicitTxn: true
 # IsSyncStatsFailed: false
-# IsWriteCacheTable: true`
+# IsWriteCacheTable: true
+# Resource_group: rg1
+# Request_unit_read: 50
+# Request_unit_write: 100.56
+# Time_queued_by_rc: 10`
```
Comment (crazycs520) [inline pkg/sessionctx/variable/session_test.go:265]:
```suggestion
# Time_queued_by_rc: 13`
```

Comment (crazycs520) [reply pkg/sessionctx/variable/session_test.go:265]: use distinct value to detect parse bug.

File: pkg/executor/slow_query_test.go [hunk 5]
```diff
@@ -138,6 +138,10 @@ func TestParseSlowLogFile(t *testing.T) {
 # Plan_from_binding: true
 # Succ: false
 # IsExplicitTxn: true
+# Resource_group: default
+# Request_unit_read: 2.158
+# Request_unit_write: 2.158
```
Comment (crazycs520) [inline pkg/executor/slow_query_test.go:143]:
```suggestion
# Request_unit_write: 2.159
```

Comment (crazycs520) [reply pkg/executor/slow_query_test.go:143]: use distinct value too.

**Comment by glorv** [comment]
@crazycs520  Thanks for the reviewing, PTAL again.

File: pkg/util/stmtsummary/v2/record.go [hunk 6]
```diff
@@ -119,6 +119,10 @@ type StmtRecord struct {
 	SumBackoffTimes      int64               `json:"sum_backoff_times"`
 	BackoffTypes         map[string]int      `json:"backoff_types"`
 	AuthUsers            map[string]struct{} `json:"auth_users"`
+	// RU
+	WRU            float64       `json:"wru"`
```
Comment (crazycs520) [inline pkg/util/stmtsummary/v2/record.go:123]: those feilds are never been used?

Comment (glorv) [reply pkg/util/stmtsummary/v2/record.go]: Fixed. Removed the partial work for adding ru in `information_schema.statements_summary`, it's should be in a separate PR.

**Review by crazycs520 — APPROVED** [review]

**Comment by glorv** [comment]
@tangenta  PTAL

**Review by tangenta — APPROVED** [review]

**Comment by glorv** [comment]
@yudongusa PTAL

**Comment by glorv** [comment]
@easonn7 PTAL

**Pull request merged**